README.md

```diff
@@ -1 +1,260 @@
+| Search Folder      | $(Common.TestResultsDirectory)                                          |
+| Run this task      | Even if a previous task has failed, unless the deployment was cancelled |
+
+Forge has created a task group that handles this, titled `Test PowerShell Module`. It contains a Pester Test Runner step, and separate Publish Test Result and Publish Code Coverage Result steps.
+
+### Code Coverage during vNext Builds
+
+If running Pester 4.0.3 or higher, you can specify a CodeCoverageOutputFile, which you can then add to the build using the `Publish Code Coverage Result` build task.
+
+This step is covered within the task group `Test PowerShell Module`.
+
+If using the Pester Test Runner build task separately, you should configure options in the Test Runner step to include the following:
+
+| Setting                   | Value                                                                                    |
+| ------------------------- | ---------------------------------------------------------------------------------------- |
+| Code Coverage Output File | $(Common.TestResultsDirectory)\Coverage-$(Build.DefinitionName)_$(Build.BuildNumber).xml |
+| Pester Version            | 4.3.1                                                                                    |
+| Force the use of a Pester Module shipped within this task | ✓ |
+
+To upload the file, you should then add the `Publish Code Coverage Result` step to your build, with the following options:
+
+| Setting            | Value                                         |
+| ------------------ | --------------------------------------------- |
+| Code coverage tool | JaCoCo                                        |
+| Summary file       | Path set in `Code Coverage Output File` above |
+
+### Packaging Artifacts
+
+Forge has created a task group that handles these steps, titled `Publish PowerShell Module`. It packages the module, and publishes the zip to the build - only publishing the nupkg if it's a release branch.
+
+#### ZIP
+
+Creating an archive of the module is simple. 
+
+```PowerShell
+[IO.DirectoryInfo]$Module = $($env:Build_StagingDirectory)\$ModuleName
+Compress-Archive -Path $Module.FullName -DestinationPath ("$($env:Build_StagingDirectory)\" + $Module.Name + '_' + $(env:Build_BuildNumber) + ".zip")
+```
+
+#### NUPKG 
+
+Creating an installable nupkg is more involved. We recommend registering a local package source temporarily, and using the PowerShellGet `Publish-Module` function to create the nupkg there.
+
+```PowerShell
+$RepoName = (New-Guid).Guid
+$Location = $($env:Build_BinariesDirectory)
+
+if (-not [bool](Get-PackageSource -Location $Location)) {
+  Register-PackageSource -Name $RepoName -Location $Location -ProviderName PowerShellGet
+}
+
+[IO.DirectoryInfo]$Module = $($env:Build_StagingDirectory)\$ModuleName
+Publish-Module -Path $Module -Repository $RepoName
```

Comment:
May want to provide some more detail if there are required modules that are a part of the publish. I know those have caused headaches for me.

Reply:
It is not. I probably tagged this in the wrong place. Not what Publish-Module requires, but if the module you're are publishing has required module dependencies that aren't marked as external dependencies, Publish-Module is going to error on you about missing sources in the repository.